packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -319,9 +319,9 @@ if (__DEV__) {
     let warningsForRoot = pendingLegacyContextWarning.get(strictRoot);
 
     if (
-      typeof instance.getChildContext === 'function' ||
       fiber.type.contextTypes != null ||
-      fiber.type.childContextTypes != null
+      fiber.type.childContextTypes != null ||
+      (instance && typeof instance.getChildContext === 'function')
```

Comment:
We try to avoid checks like `something &&` in terms of explicit `something !== null &&` when that argument is coming from our own code (and we know it's either `null` or an object).